### PR TITLE
WiP: APK 3.0-pre0

### PR DESCRIPTION
**Pull request by dangowrt**
Maintainer: @aparcar 
Compile tested: x86/64
Run tested: x86/64 (QEMU)

Description:
Get work-in-progress APK to build here, just to use CI.

### CODE REVIEW

**Comment by aparcar** [comment]
Looks overall good to me. I'm assuming it was not possible to make it work with the Makefile and therefore meson is required?

My issue is that I want APK in core for obvious reasons and there is still no clear timeline for Meson over there. Since I'm not working on Meson I'd rather stay with Makefile.

I'm assuming you reported all patches upstream or are they to OpenWrt specific?

File: utils/apk/Makefile [hunk 1]
```diff
@@ -16,7 +16,11 @@ PKG_LICENSE_FILES:=LICENSE
 PKG_INSTALL:=1
 PKG_BUILD_PARALLEL:=1
 
+HOST_BUILD_DEPENDS:=lua/host lua-lzlib/host meson/host
```
Comment (aparcar) [inline utils/apk/Makefile:19]: Meson doesn't seem to be a requirement anymore?

File: utils/apk/Makefile [hunk 2]
```diff
@@ -16,13 +21,17 @@ PKG_LICENSE_FILES:=LICENSE
 PKG_INSTALL:=1
 PKG_BUILD_PARALLEL:=1
 
+HOST_BUILD_DEPENDS:=lua/host lua-lzlib/host
+PKG_BUILD_DEPENDS:=$(HOST_BUILD_DEPENDS)
+
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/host-build.mk
```
Comment (neheb) [inline utils/apk/Makefile:28]: Now that meson is in base, might as well use it here.

**Comment by aparcar** [comment]
Should we close this PR and work together on https://github.com/openwrt/openwrt/pull/4294 ?

**Comment by neheb** [comment]
Yes.

**Comment by neheb** [comment]
Closing this as this is moving and being developed in base.

**Pull request closed without merging**